Tests/ConfidenceTests/ConfidenceTest.swift

```diff
@@ -318,6 +318,50 @@ class ConfidenceTest: XCTestCase {
         XCTAssertEqual(flagApplier.applyCallCount, 1)
     }
 
+    func testAwaitReconciliation() async throws {
+        class FakeClient: XCTestCase, ConfidenceResolveClient {
+            var resolveStats: Int = 0
+            var resolvedValues: [ResolvedValue] = []
+            func resolve(ctx: ConfidenceStruct) async throws -> ResolvesResult {
+                self.resolveStats += 1
+                return .init(resolvedValues: resolvedValues, resolveToken: "token")
+            }
+        }
+
+        let client = FakeClient()
+        client.resolvedValues = [
+            ResolvedValue(
+                variant: "control",
+                value: .init(structure: ["size": .init(integer: 3)]),
+                flag: "flag",
+                resolveReason: .match)
+        ]
+
+        let confidence = Confidence.Builder(clientSecret: "test")
+            .withContext(initialContext: ["targeting_key": .init(string: "user2")])
+            .withFlagResolverClient(flagResolver: client)
+            .withFlagApplier(flagApplier: flagApplier)
+            .build()
+        Task {
+            await confidence.putContext(context: ["hello": .init(string: "world")])
+        }
+        try await Task.sleep(nanoseconds: 100 * 1000) // 100 ms wait
+        await confidence.awaitReconciliation()
```

Comment:
It's impossible to know if a `putContext` from another thread (Task) has been started or not. Even for a real application, the `awaitReconciliation()` could always be called "too soon" with respect to more putContexts. In fact, I couldn't come up with a programmatic way to test this that doesn't include a sleep

Reply:
In other words, this function would be more useful if `putContext` wasn't awaitable. I am now unsure if having both awaitable makes sense